modules/data_merge/data_merge_worker.py

```diff
@@ -27,28 +41,64 @@ def data_merge_worker(
     detection_input_queue, odometry_input_queue, output_queue are data queues.
     controller is how the main process communicates to this worker process.
     """
+    if merge_data_type == "OBSTACLES":
+        merge_data_type = MergeDataType.OBSTACLES
+    elif merge_data_type == "DETECTIONS":
+        merge_data_type = MergeDataType.DETECTIONS
+
     detections = []
+    obstacles = []
     while not controller.is_exit_requested():
         controller.check_pause()
 
-        try:
-            detection: lidar_detection.LidarDetection = detection_input_queue.queue.get_nowait()
-            detections.append(detection)
-        except queue.Empty:
-            time.sleep(delay)
+        if merge_data_type == MergeDataType.OBSTACLES:
+            try:
+                new_obstacle: obstacle.Obstacle = obstacle_input_queue.queue.get_nowait()
+                obstacles.append(new_obstacle)
+            except queue.Empty:
+                if len(obstacles) == 0:
+                    continue
+                time.sleep(delay)
 
-        try:
-            odometry: drone_odometry_local.DroneOdometryLocal = (
-                odometry_input_queue.queue.get_nowait()
+            try:
+                odometry: drone_odometry_local.DroneOdometryLocal = (
+                    odometry_input_queue.queue.get_nowait()
+                )
+            except queue.Empty:
+                continue
```

Comment:
since this must happen not matter the merge data type, you should move it outside after the if statement